src/main/java/com/denizenscript/depenizen/bukkit/objects/luckperms/LuckPermsGroupTag.java

```diff
@@ -0,0 +1,160 @@
+
+import com.denizenscript.denizencore.objects.Fetchable;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.tags.Attribute;
+import com.denizenscript.denizencore.tags.ObjectTagProcessor;
+import com.denizenscript.denizencore.tags.TagContext;
+import com.denizenscript.depenizen.bukkit.bridges.LuckPermsBridge;
+import net.luckperms.api.model.group.Group;
+
+import java.util.OptionalInt;
+
+public class LuckPermsGroupTag implements ObjectTag {
+
+    // <--[ObjectType]
+    // @name LuckPermsGroupTag
+    // @prefix luckpermsgroup
+    // @base ElementTag
+    // @format
+    // The identity format for group is <group_name>
+    // For example, 'luckpermsgroup@my_group'.
+    //
+    // @plugin Depenizen, LuckPerms
+    // @description
+    // A LuckPermsGroupTag represents a LuckPerms group.
+    //
+    // -->
+
+    @Fetchable("luckpermsgroup")
+    public static LuckPermsGroupTag valueOf(String string, TagContext context) {
+        if (string.startsWith("luckpermsgroup@")) {
+            string = string.substring("@luckpermsgroup".length());
+        }
+        Group group = LuckPermsBridge.luckPermsInstance.getGroupManager().getGroup(string);
+        if (group == null) {
+            return null;
+        }
+        return new LuckPermsGroupTag(group);
+    }
+
+    public static boolean matches(String arg) {
+        if (arg.startsWith("luckpermsgroup@")) {
+            return true;
+        }
+        return LuckPermsBridge.luckPermsInstance.getGroupManager().isLoaded(arg);
+    }
+
+    Group group;
+
+    public LuckPermsGroupTag(Group group) {
+        this.group = group;
+    }
+
+    String prefix = "LuckPermsGroup";
+
+    @Override
+    public String getPrefix() {
+        return prefix;
+    }
+
+    @Override
+    public ObjectTag setPrefix(String prefix) {
+        this.prefix = prefix;
+        return this;
+    }
+
+    @Override
+    public boolean isUnique() {
+        return true;
+    }
+
+    @Override
+    public String identify() {
+        return "luckpermsgroup@" + group.getName();
+    }
+
+    @Override
+    public String identifySimple() {
+        return identify();
+    }
+
+    @Override
+    public String toString() {
+        return identify();
+    }
+
+    @Override
+    public ObjectTag getObjectAttribute(Attribute attribute) {
+        return tagProcessor.getObjectAttribute(this, attribute);
+    }
+
+    public Group getGroup() {
+        return group;
+    }
+
+    public static void register() {
+
+        // <--[tag]
+        // @attribute <LuckPermsGroupTag.name>
+        // @returns ElementTag
+        // @plugin Depenizen, LuckPerms
+        // @description
+        // Returns the group's name.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "name", (attribute, object) -> {
+            return new ElementTag(object.getGroup().getName(), true);
+        });
```

Comment:
Correct me if I'm wrong, but I don't think LP groups can be renamed (i.e. this can be static)